_posts/2019-07-24-arc.adoc

```diff
@@ -0,0 +1,188 @@
+---
+layout: post
+date:   2019-07-24 00:00 +0100
+author: mkouba
+synopsis: Quarkus ArC is a build-time oriented dependency injection based on CDI 2.0. But what does it actually mean and what benefits does a build-time processing DI bring?
+---
+
+= Quarkus Dependency Injection
+
+Quarkus ArC is a build-time oriented dependency injection based on http://cdi-spec.org/[CDI 2.0].
+In this blogpost, we're going to explain the relationship to the specification and describe some of the benefits and drawbacks of the build-time processing design.
+
+== Compatibility
+
+It does not make sense to reinvent the wheel, especially when it comes to dependency injection.
+Threre are many frameworks that try to solve similar problems.
+A year ago, we made a design decision to build Quarkus DI on top of CDI.
+We had some very good reasons to choose CDI:
+
+1. CDI is a mature and proven component model
+2. We at Red Hat have almost ten years of experience with developing http://weld.cdi-spec.org/[Weld - the CDI Reference Implementation]
+3. CDI API is built on top of `javax.inject` so that it should be easy to migrate from any DI framework compatible with `@Inject`
```

Comment:
```suggestion
3. The CDI API is built on top of `javax.inject` so that it should be easy to migrate from any DI framework compatible with `@Inject`
```
maybe?